src/deluge/io/midi/device_specific/midi_device_lumi_keys.h

```diff
@@ -0,0 +1,194 @@
+	    {0x40, 0x00}, {0x60, 0x00}, {0x00, 0x01}, {0x20, 0x01}, {0x40, 0x01}, {0x60, 0x01}, {0x00, 0x02}, {0x20, 0x02},
+	    {0x40, 0x02}, {0x60, 0x02}, {0x00, 0x03}, {0x20, 0x03}, {0x40, 0x03}, {0x60, 0x03}, {0x00, 0x04} // Ch. 16
+	};
+
+	enum class RootNote { C = 0, C_SHARP, D, D_SHARP, E, F, F_SHARP, G, G_SHARP, A, A_SHARP, B };
+
+	static constexpr uint8_t sysexRootNoteCodes[MIDI_DEVICE_LUMI_KEYS_KEY_COUNT][2] = {
+	    {0x03, 0x00}, // C
+	    {0x23, 0x00}, // C#
+	    {0x43, 0x00}, // D
+	    {0x63, 0x00}, // D#
+	    {0x03, 0x01}, // E
+	    {0x23, 0x01}, // F
+	    {0x43, 0x01}, // F#
+	    {0x63, 0x01}, // G
+	    {0x03, 0x02}, // G#
+	    {0x23, 0x02}, // A
+	    {0x43, 0x02}, // A#
+	    {0x63, 0x02}  // B
+	};
+
+	enum class MIDIMode { MPE = 0, MULTI, SINGLE };
+
+	static constexpr uint8_t sysexMidiModeCodes[3] = {
+	    0x20, // MPE
+	    0x00, // Multi
+	    0x40  // Single
+	};
+
+	enum class MPEZone { LOWER = 0, UPPER };
+
+	static constexpr uint8_t SysexMpeZoneCodes[2] = {
+	    0x05, // LOWER
+	    0x25  // UPPER
+	};
+
+	enum class Scale {
+		MAJOR = 0, // IONIAN
+		MINOR,     // AEOLIAN
+		HARMONIC_MINOR,
+		PENTATONIC_NEUTRAL,
+		PENTATONIC_MAJOR,
+		PENTATONIC_MINOR,
+		BLUES,
+		DORIAN,
+		PHRYGIAN,
+		LYDIAN,
+		MIXOLYDIAN,
+		LOCRIAN,
+		WHOLE_TONE,
+		ARABIC_A,
+		ARABIC_B,
+		JAPANESE,
+		RYUKYU,
+		EIGHT_TONE_SPANISH,
+		CHROMATIC
+	};
+
+	static constexpr uint16_t scaleNotes[MIDI_DEVICE_LUMI_KEYS_SCALE_COUNT] = {
+	    MIDI_DEVICE_LUMI_SCALE_MAJOR,
+	    MIDI_DEVICE_LUMI_SCALE_MINOR,
+	    MIDI_DEVICE_LUMI_SCALE_HARMONIC_MINOR,
+	    MIDI_DEVICE_LUMI_SCALE_PENTATONIC_NEUTRAL,
+	    MIDI_DEVICE_LUMI_SCALE_PENTATONIC_MAJOR,
+	    MIDI_DEVICE_LUMI_SCALE_PENTATONIC_MINOR,
+	    MIDI_DEVICE_LUMI_SCALE_BLUES,
+	    MIDI_DEVICE_LUMI_SCALE_DORIAN,
+	    MIDI_DEVICE_LUMI_SCALE_PHRYGIAN,
+	    MIDI_DEVICE_LUMI_SCALE_LYDIAN,
+	    MIDI_DEVICE_LUMI_SCALE_MIXOLYDIAN,
+	    MIDI_DEVICE_LUMI_SCALE_LOCRIAN,
+	    MIDI_DEVICE_LUMI_SCALE_WHOLE_TONE,
+	    MIDI_DEVICE_LUMI_SCALE_ARABIC_A,
+	    MIDI_DEVICE_LUMI_SCALE_ARABIC_B,
+	    MIDI_DEVICE_LUMI_SCALE_JAPANESE,
+	    MIDI_DEVICE_LUMI_SCALE_RYUKYU,
+	    MIDI_DEVICE_LUMI_SCALE_8TONE_SPANISH,
+	    MIDI_DEVICE_LUMI_SCALE_CHROMATIC};
+
+	// By reference to presetScaleNames in global scope
+	static constexpr Scale delugeScaleMap[9] = {
+	    Scale::MAJOR,      // MAJOR (IONIAN) -> MAJOR
+	    Scale::MINOR,      // MINOR (AEOLIAN) -> MINOR
+	    Scale::DORIAN,     // DORIAN -> DORIAN
+	    Scale::PHRYGIAN,   // PHRYGIAN -> PHRYGIAN
+	    Scale::LYDIAN,     // LYDIAN -> LYDIAN
+	    Scale::MIXOLYDIAN, // MIXOLYDIAN -> MIXOLYDIAN,
+	    Scale::LOCRIAN,    // LOCRIAN -> LOCRIAN
+	    Scale::CHROMATIC,  // RANDOM -> CHROMATIC (no idea how else to handle)
+	    Scale::CHROMATIC   // NONE -> CHROMATIC
+	};
+
+	static constexpr uint8_t sysexScaleCodes[MIDI_DEVICE_LUMI_KEYS_SCALE_COUNT][2] = {
+	    {0x02, 0x00}, {0x22, 0x00}, {0x42, 0x00}, {0x62, 0x00}, {0x02, 0x01}, {0x22, 0x01}, {0x42, 0x01},
```

Comment:
scale code << 5 + 2

Reply:
Yeah fixed this through algorithmic generation as of the push just now.